src/containers/TokenUpload/index.ts

```diff
@@ -9,32 +14,26 @@ import { State } from 'types'
 
 const mapState = ({
   ipfs: {
-    fileBuffer,
-    oFile,
     fileHash,
     filePath,
-    fileContent,
     json,
   },
   tokenList: {
     customTokenList,
   },
 }: Partial<State>) => ({
-  fileBuffer,
-  oFile,
   fileHash,
   filePath,
-  fileContent,
   json,
   customTokenList,
+  potentiallyValidHash: fileHash.length == 46,
 })
 
 export default connect(
   mapState,
   {
-    getFileContentFromIPFS,
-    openModal,
+    setIPFSFileHash,
     setNewIPFSCustomListAndUpdateBalances,
     setTokenListType,
-    setUploadFileParams,
+    openModal,
   })(IPFSHOC(TokenUpload) as any)
```

Comment:
`TokenUpload` is wrapped in `IPFSHOC` twice, here and in `TokenUpload` component export

Reply:
Still getting timeouts on seemingly valid hashes.

Reply:
We may need to account for `hash+path` in input
like `QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG/readme`, leading to https://gateway.ipfs.io/ipfs/QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG/readme

Reply:
good point about the `path`ing but I am not really sure rn what do think on that